src/main/java/com/stripe/net/Webhook.java

```diff
@@ -47,10 +48,34 @@ public static Event constructEvent(String payload, String sigHeader, String secr
   public static Event constructEvent(
       String payload, String sigHeader, String secret, long tolerance)
       throws SignatureVerificationException {
-    Event event =
-        StripeObject.deserializeStripeObject(
-            payload, Event.class, ApiResource.getGlobalResponseGetter());
-    Signature.verifyHeader(payload, sigHeader, secret, tolerance);
+    return constructEvent(payload, sigHeader, secret, tolerance, null);
+  }
+
+  /**
+   * Returns an Event instance using the provided JSON payload. Throws a JsonSyntaxException if the
+   * payload is not valid JSON, and a SignatureVerificationException if the signature verification
+   * fails for any reason.
+   *
+   * @param payload the payload sent by Stripe.
+   * @param sigHeader the contents of the signature header sent by Stripe.
+   * @param secret secret used to generate the signature.
+   * @param tolerance maximum difference in seconds allowed between the header's timestamp and the
+   *     current time
+   * @param clock instance of clock if you want to use custom time instance
+   * @return the Event instance
+   * @throws SignatureVerificationException if the verification fails.
+   */
+  public static Event constructEvent(String payload, String sigHeader, String secret, long tolerance, Clock clock)
+    throws SignatureVerificationException {
+    return constructInternalEvent(payload, sigHeader, secret, tolerance, clock);
+  }
+
+  private static Event constructInternalEvent(String payload, String sigHeader, String secret, long tolerance, Clock clock)
```

Comment:
Can we inline this method into the longest overload of `constructEvent`?

Reply:
updated, thank you